base/Dockerfile

```diff
@@ -16,7 +16,7 @@ RUN cd $GOPATH/src \
 
 # Build NF (AMF, AUSF, N3IWF, NRF, NSSF, PCF, SMF, UDM, UDR)
 RUN cd $GOPATH/src/free5gc/src \
-    && for d in * ; do if [ -f "$d/$d.go" ] ; then go build -o ../bin/"$d" -x "$d/$d.go" ; fi ; done ;
+    && for d in * ; do if [ -f "$d/$d.go" ] ; then CGO_ENABLED=0 go build -a -installsuffix nocgo -o ../bin/"$d" -x "$d/$d.go" ; fi ; done ;
```

Comment:
What does these tag means?

Reply:
I used to copy paste that set of flags when compiling statically linked go code until you asked the question and thank you for that :+1: 

So, dynamic linking comes from the use of `cgo` which is according to [[1]](https://golang.org/src/cmd/cgo/doc.go):
> Cgo enables the creation of Go packages that call C code.

This is done either by doing an `import "C"` or by calling a go package that does.
  - `CGO_ENABLED=0`: By default, go builds statically linked binaries. This behaviour can be controlled by the `CGO_ENABLED` environment variable. When it set to 0, it disables `cgo` which is enabled by default for native builds [[1]](https://golang.org/src/cmd/cgo/doc.go).

  - `-installsuffix nocgo`: I find out that the `-installsuffix nocgo` flag was a workaround for old versions of Go (<1.10). For the current version of Go it has nothing to do with linking [[2]](https://golang.org/pkg/cmd/go/). So we can safely remove it.

  - `-a`: this flag is used to tell `go build` to recompile all packages and all their dependencies [[2]](https://golang.org/pkg/cmd/go/) [[3]](https://github.com/golang/go/commit/8c3005c4929e26cba74c80703123e150fa3fda1a)

Static linking in Golang is still an open question and there is some proposals to make it more simple and user friendly [[4]](https://github.com/golang/go/issues/26492). A fix was originally planned for the 1.14 release [[6]](https://github.com/golang/go/issues/26492#issuecomment-500254614).

Finally, the new way of doing things in linux seems to be using this set of flags: `-tags netgo -ldflags '-extldflags "-static"'` [[5]](https://github.com/golang/go/issues/26492#issuecomment-435462350), where:
  - `-extldflags "-static"` is used to pass the `-static` flag to the external linker [[7]](https://golang.org/doc/go1.5)
  - `-tags netgo` is a build tag used to enforce the use of the Go resolver, not the cgo one [[7]](https://golang.org/doc/go1.5)
  - `-ldflags` is flag to wrap all the others and pass them on each go tool link invocation [[7]](https://golang.org/doc/go1.5)

I will update the build with these new flags.

#### References
[1] https://golang.org/src/cmd/cgo/doc.go
[2] https://golang.org/pkg/cmd/go/
[3] https://github.com/golang/go/commit/8c3005c4929e26cba74c80703123e150fa3fda1a
[4] https://github.com/golang/go/issues/26492
[5] https://github.com/golang/go/issues/26492#issuecomment-435462350
[6] https://github.com/golang/go/issues/26492#issuecomment-500254614
[7] https://golang.org/doc/go1.5

Reply:
sounds great!